### ISSUE TITLE
ActionItems lacks proper support for VoiceOver on iOS

### ISSUE THREAD

**Issue by m-abs** [NativeScript/NativeScript#2796]
The text-property on `ActionItem`s on the `ActionBar` isn't read by `VoiceOver` on `iOS`, only the icon name is and `ActionItem` doesn't have the property `automationText`. That's because `ActionItem` doesn't inherit from `ui/core/view`.

On iOS `UIBarButtonItem` is used to create the `ActionItem`, but `UIBarButtonItem` doesn't implement the `UIAccessibility`-protocol, but according to this stackoverflow http://stackoverflow.com/questions/23800764/ios-7-uibarbuttonitem-with-accessibility it should be possible it activate it anyways.

### Which platform(s) does your issue occur on?
iOS

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.3.0
- Cross-platform modules: 2.3.0
- Runtime(s): tns-ios 2.3.0

### Please tell us how to recreate the issue in as much detail as possible.
* Create an `ActionBar` with an `ActionItem` set a text-label on the ActionItem
* Run the app on iOS with VoiceOver enabled
* Listen to the text being read, it's the icon-name followed by "button"

**Comment by m-abs** [comment]
# Update:
I think I see the problem:
https://github.com/NativeScript/NativeScript/blob/2.3.0/tns-core-modules/ui/action-bar/action-bar.ios.ts#L134-L163

If we have an icon on the `ActionItem` the text is never set on the `UIBarButtonItem`, so of cause iOS' VoiceOver can't read the text.

I'm working on a PR that hopefully solves the problem.

**Comment by vchimev** [comment]
Hi @m-abs,

Thank you for the pull request.

Could you please send us a sample page which demonstrates the issue along with:
- how does VoiceOver read it currently and
- how is it expected to read it?

In this way, we would be able to review your pull request properly.

For example, my test page is as follow:
```
<Page>
  <Page.actionBar>
    <ActionBar >
      <ActionBar.actionItems>
        <ActionItem text="Item" />
        <ActionItem>
          <ActionItem.actionView>
            <Button text="View" automationText="test" />
          </ActionItem.actionView>
        </ActionItem>
      </ActionBar.actionItems>
    </ActionBar>
  </Page.actionBar>
  <StackLayout>
    <Label text="This is a test."/>
  </StackLayout>
</Page>
```
It is read as follows: *Item -> Button, Test -> Button*

**Comment by m-abs** [comment]
HI @vchimev,

The problem is caused by adding an icon to the `ActionItem`, this way the text is never added to the `UINavigationButton`.

I've written a demo as requested, it's based on `tns create APP_NAME --template typescript` and runs with a local install of `tns-core-modules` with the patch.

The screenshots is from running on the iPhone emulator, so you can see the `Accessibility Inspector`:

The ActionItem on https://github.com/m-abs/actionitem-demo/blob/master/app/main-page.xml#L5

Here without my patch:
<img width="378" alt="screen shot 2016-10-01 at 17 19 28" src="https://cloud.githubusercontent.com/assets/1348705/19015015/ebbc9d3a-87fb-11e6-988d-7074eb7e08de.png">
As you can see the inspector says the Label is **X** the filename of my icon.

Expected reading order is "My ActionItem Text" -> "Button"
As seen here with the patch:
<img width="374" alt="screen shot 2016-10-01 at 17 28 32" src="https://cloud.githubusercontent.com/assets/1348705/19015040/7b4ee16a-87fc-11e6-92c2-63a6224f578c.png">

If you were testing the `ActionItem` > `Button` on `tns-core-modules@2.3.0`, my other PR for `automationText` on `iOS` weren't a part of that release. (https://github.com/NativeScript/NativeScript/pull/2678) but it is on master, so it's in my current PR.

Expected reading order "AutoText" > "Button"
As seen here:
<img width="372" alt="screen shot 2016-10-01 at 17 31 50" src="https://cloud.githubusercontent.com/assets/1348705/19015060/f7d73944-87fc-11e6-8fcb-b3cafb945569.png">

**Note:** that `automationText` doesn't exist on `ActionItem` because `contentDescription` doesn't exist on the `MenuItem` used on `Android`.

**Comment by vchimev** [comment]
@m-abs, many thanks for the details.
